libzhlgen/test/functions/mac/1.6.13/Ship.zhl

```diff
@@ -0,0 +1,73 @@
+"NOT LISTED IN GHIDRA":
+cleanup __amd64 Globals::Ellipse Ship::GetBaseEllipse(Ship *this); /* This comment here wasn't made by Dino ---> /* TODO: Ellipse is actually returned on RAX:XMM0 or maybe it's RDX:XMM0 */
+"554889e5488b871805000048":
+cleanup __amd64 bool Ship::RoomLocked(Ship *this, int roomId);
+"554889e541574156534883ec18f30f1145e44889":
+cleanup __amd64 void Ship::OnRenderJump(Ship *this, float progress);
+"554889e5415741564155415453504189f64889fb":
+cleanup __amd64 void Ship::OnRenderBase(Ship *this, bool unk);
+"554889e54157415653504189f64989ff4180bf10":
+cleanup __amd64 void Ship::OnRenderFloor(Ship *this, bool unk);
+"554889e54156534989fe498b461049394618742dbb0100000031c90f1f440000488b3cc8e83715ff":
+cleanup __amd64 void Ship::OnRenderSparks(Ship *this);
+"554889e54157415653504989fe498b464049394648744cbb":
+cleanup __amd64 void Ship::OnRenderBreaches(Ship *this);
+"554889e54157415641554154534883ec284189d74189f448":
+cleanup __amd64 void Ship::BreachSpecificHull(Ship *this, int grid_x, int grid_y);
+"554889e54157415653504189d64989f74c89fb48c1eb":
+cleanup __amd64 int Ship::GetSelectedRoomId(Ship *this, int x, int y, bool unk);
+"554889e54157415653504189f64989ff498b4710":
+cleanup __amd64 void Ship::SetSelectedRoom(Ship *this, int roomId);
+".554889e54863c6488b4f10488b3cc10fb6f2":
+cleanup __amd64 void Ship::SetRoomBlackout(Ship *this, int roomId, bool blackout);
+".554889e54863c6488b4f10488b3cc10fb6f2":
+cleanup __amd64 int Ship::GetAvailableRoomSlot(Ship *this, int roomId, bool intruder);
+"554889e54157415653504989fe4c63fe":
+cleanup __amd64 int Ship::EmptySlots(Ship *this, int roomId);
+"554889e585f67823":
+cleanup __amd64 bool Ship::FullRoom(Ship *this, int roomId, bool intruder);
+"554889e54157415641554154534883ec2889d341":
+cleanup __amd64 int Ship::GetAvailableRoom(Ship *this, int preferred, bool intruder);
+"554889e54157415641554154534883ec184189d78975d449":
+cleanup __amd64 void Ship::OnRenderWalls(Ship *this, bool forceView, bool doorControlMode);
+"554889e54157415641554154534883ec284989f64989ff498d9ff004":
+cleanup __amd64 void Ship::OnLoop(Ship *this, std::vector<float>& oxygenLevels);
+"554889e54157415641554154534883ec688975cc":
+cleanup __amd64 void Ship::BreachRandomHull(Ship *this, int roomId);
+"554889e54157415641554154534883ec184989f74989fd49":
+cleanup __amd64 std::vector<Repairable*> Ship::GetHullBreaches(Ship *this, bool onlyDamaged);
+"554889e54157415641554154534881ec980100000f294580":
+cleanup __amd64 void Ship::LockdownRoom(Ship *this, int roomId, Pointf pos);
+"554889e54157415641554154534881ec88030100":
+cleanup __amd64 void Ship::OnInit(Ship *this, ShipBlueprint* bp);
+"UNKNOWN FUNCTION DEFENITION":
+noop void* Global__NOP__Ship_OnInit_4;
```

Comment:
These will have to be very manually found and they *might* need more or less of them.

Talk to me sometime specifically about this, but basically I deleted very specific operations from the built-in game's door placement code so it doesn't crash with out-of-range errors. These affected Linux at anything larger than 2x2 but just happened to not crash Windows until like 14x14 because of how the memory was laid out.

We can study the Mac one and specifically no-op the correct bytes or we could reverse engineer & rewrite the entire Ship::OnInit function and remove this no-op nonsense from all the versions 😄